### ISSUE TITLE
(proposal) feat(semantic): add `ReferenceFlag::JSDoc`

### ISSUE THREAD

**Issue by DonIsaac** [oxc-project/oxc#4443]
Consider this code snippet:
```ts
import type { RelevantType } from './foo'

/**
 * @see {@link RelevantType}
 */
type OtherType = number
```
It would be nice if `no-unused-vars` did not report `RelevantType` as unused. Maybe we could mark this link as a usage?

**Comment by Boshen** [comment]
jsdoc isn't part of the language :thinking: I'm going to reject this unless something more convincing comes up.
